projects/packages/search/src/dashboard/components/module-control/index.jsx

```diff
@@ -0,0 +1,235 @@
+	'Instant search will allow your visitors to get search results as soon as they start typing. If deactivated, Jetpack Search will still optimize your search results but visitors will have to submit a search query before seeing any results.',
+	'jetpack'
+);
+const RETURN_PATH = 'admin.php?page=jetpack-search';
+const SEARCH_CUSTOMIZE_URL = 'admin.php?page=jetpack-search-configure';
+const WIDGETS_EDITOR_URL = 'customize.php?autofocus[panel]=widgets&return=%s';
+
+/**
+ * Search settings component to be used within the Performance section.
+ *
+ * @param {object} props - Component properties.
+ * @param {string} props.domain - Calypso slug.
+ * @param {string} props.siteAdminUrl - site admin URL.
+ * @param {string} props.upgradeBillPeriod - billing cycle for upgrades.
+ * @param {Function} props.updateOptions - function to update settings.
+ * @param {boolean} props.isSavingEitherOption - true if Saving options.
+ * @param {boolean} props.isModuleEnabled - true if Search module is enabled.
+ * @param {boolean} props.isInstantSearchEnabled - true if Instant Search is enabled.
+ * @param {boolean} props.isInstantSearchPromotionActive - true if search promotion is active.
+ * @param {boolean} props.supportsOnlyClassicSearch - true if site has plan that supports only Classic Search.
+ * @param {boolean} props.supportsSearch - true if site has plan that supports either Classic or Instant Search.
+ * @param {boolean} props.supportsInstantSearch - true if site has plan that supports Instant Search.
+ * @param {boolean} props.isTogglingModule - true if toggling Search module.
+ * @param {boolean} props.isTogglingInstantSearch - true if toggling Instant Search option.
+ * @returns {React.Component}	Search settings component.
+ */
+export default function SearchModuleControl( {
+	siteAdminUrl,
+	updateOptions,
+	domain,
+	isSavingEitherOption,
+	isModuleEnabled,
+	isInstantSearchEnabled,
+	isInstantSearchPromotionActive,
+	upgradeBillPeriod,
+	supportsOnlyClassicSearch,
+	supportsSearch,
+	supportsInstantSearch,
+	isTogglingModule,
+	isTogglingInstantSearch,
+} ) {
+	const isInstantSearchCustomizeButtonDisabled =
+		isSavingEitherOption ||
+		! isModuleEnabled ||
+		! isInstantSearchEnabled ||
+		! supportsInstantSearch;
+	const isWidgetsEditorButtonDisabled = isSavingEitherOption || ! isModuleEnabled;
+	const returnUrl = encodeURIComponent( siteAdminUrl + RETURN_PATH );
+	const upgradeUrl = getRedirectUrl(
+		upgradeBillPeriod === 'monthly' ? 'jetpack-search-monthly' : 'jetpack-search',
+		{ site: domain }
+	);
+
+	const toggleSearchModule = useCallback( () => {
+		const oldOption = {
+			module_active: isModuleEnabled,
+			instant_search_enabled: isInstantSearchEnabled,
+		};
+		const newOption = {
+			module_active: ! isModuleEnabled,
+		};
+		if ( isInstantSearchEnabled !== ! isModuleEnabled ) {
+			newOption.instant_search_enabled = ! isModuleEnabled && supportsInstantSearch;
+		}
+		updateOptions( newOption, oldOption );
```

Comment:
Interesting; why does `updateJetpackSettings` require two parameters?

Reply:
Good point. Could definitely use only one. Didn't want to use states in the `updateJetpackSettings`, but it probably be better to just use it. Addressed in 09295c3c83